### PR TITLE
Fix elastic transformation if only partially applied

### LINKED ISSUE #2301: `NotImplementedError` for elastic transformation with probability p < 1
### Describe the bug

With the newest kornia release (0.6.11), the random elastic transformation fails if it is not applied to every image in the batch.

The problem is that the `apply_non_transform_mask()` method in `_AugmentationBase` per default raises an `NotImplementedError` and since this method is not overwritten in `RandomElasticTransform`, the error is raised. I see that for the other `apply_non*` methods the default is to just return the input.

I see two different solutions:
1. Change the default for `apply_non_transform_mask` to return the input in `_AugmentationBase`.
2. Overwrite the method in `RandomElasticTransform` and just return the input there.

There might be good reasons to keep the `NotImplementedError` in the base class, therefore I wanted to ask first what solution you prefer. I could make a PR for this.

### Reproduction steps

```python
import torch
import kornia.augmentation as K

features = torch.rand(5, 100, 480, 640, dtype=torch.float32, device="cuda")
labels = torch.randint(0, 10, (5, 1, 480, 640), dtype=torch.int64, device="cuda")

torch.manual_seed(0)
aug = K.AugmentationSequential(
    K.RandomElasticTransform(alpha=(0.7, 0.7), sigma=(16, 16), padding_mode="reflection", p=0.2)
)

features_transformed, labels_transformed = aug(features, labels.float(), data_keys=["input", "mask"])
```

### Expected behavior

No `NotImplementedError`.

### Environment

```shell
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```
- PyTorch Version (e.g., 1.0): 2.0
- OS (e.g., Linux): Linux
- How you installed PyTorch (`conda`, `pip`, source): pip
- Build command you used (if compiling from source):
- Python version: 3.10.9
- CUDA/cuDNN version: 11.8
- GPU models and configuration: 3090
- Any other relevant information:
```

### PR DESCRIPTION
**Pull request by JanSellner**
#### Changes

Fixes #2301 

I also removed the argument docstring for the recently removed `mode` parameter.

#### Type of change

- [x] 📚  Documentation Update
- [x] 🧪 Tests Cases
- [x] 🐞 Bug fix (non-breaking change which fixes an issue)

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: test/augmentation/test_augmentation.py [hunk 1]
```diff
@@ -3704,6 +3705,28 @@ def test_mask_transform(self, device, dtype):
...
+
+        aug_list = AugmentationSequential(RandomElasticTransform())
+        features = torch.rand(2, 3, 10, 10, dtype=dtype, device=device)
+        labels = torch.randint(0, 10, (2, 1, 10, 10), dtype=dtype, device=device)
+
+        # Make sure the transformation works correctly even if only applied to some images in the batch
+        to_apply = torch.tensor(batch_prob, device=device)
+        with patch.object(aug_list[0], '__batch_prob_generator__', return_value=to_apply):
+            features_transformed, labels_transformed = aug_list(features, labels, data_keys=["input", "mask"])
+            assert torch.all(aug_list._params[0].data["batch_prob"] == to_apply)
```
Comment (johnnv1) [inline test/augmentation/test_augmentation.py:3720]:
```suggestion
            assert_close(aug_list._params[0].data["batch_prob"], to_apply)
```

File: test/augmentation/test_augmentation.py [hunk 2]
```diff
@@ -3704,6 +3705,28 @@ def test_mask_transform(self, device, dtype):
...
+
+            for b, applied in enumerate(batch_prob):
+                if applied:
+                    assert features_transformed[b].ne(features).any()
+                    assert labels_transformed[b].ne(labels).any()
+                else:
+                    assert features_transformed[b].eq(features[b]).all()
+                    assert labels_transformed[b].eq(labels[b]).all()
```
Comment (johnnv1) [inline test/augmentation/test_augmentation.py:3728]:
```suggestion
                    assert_close(features_transformed[b], features[b])
                    assert_close(labels_transformed[b], labels[b])
```

Comment (johnnv1) [inline test/augmentation/test_augmentation.py:3728]: I think we should have a expected tensor, and just use the batch prob as index to assert the cases

Comment (JanSellner) [reply test/augmentation/test_augmentation.py]: I replaced the loop with a batch prob index access approach. Is this what you meant?

**Review by johnnv1 — APPROVED** [review]
LGTM

**Review by edgarriba — APPROVED** [review]
